### PR TITLE
Additional Review Comments

### PR DESCRIPTION
**Pull request by linuxwolf**
Addressing extra review comments from @annevk and @codehag.  This is still a work in progress.

### CODE REVIEW

File: javascript-mjs/draft.md [hunk 1]
```diff
@@ -149,51 +149,43 @@ Implementations need to determine a character encoding scheme in order to decode
 
 How implementations determine the character encoding scheme can be subject to processing rules that are out of the scope of this document.  For example, transport protocols can require that a specific character encoding scheme is to be assumed if the optional charset parameter is not specified, or they can require that the charset parameter is used in certain cases.  Such requirements are not considered part of this document.
 
-Implementations that support binary source text MUST support binary source text encoded using the UTF-8 {{RFC3629}} character encoding scheme.  Other character encoding schemes MAY be supported.  Use of UTF-8 to encode binary source text is encouraged but not required.
+Implementations that support binary source text MUST support binary source text encoded using the UTF-8 {{RFC3629}} character encoding scheme.  Module goal sources MUST be encoded as UTF-8, all other encoding will fail.  Source goal sources SHOULD be encoded as UTF-8; other character encoding schemes MAY be supported, but are discouraged.
```
Comment (annevk) [inline javascript-mjs/draft.md:152]:
```suggestion
Implementations that support binary source text MUST support binary source text encoded using the UTF-8 {{RFC3629}} character encoding scheme.  Module goal sources MUST be encoded as UTF-8, all other encodings will fail.  Source goal sources SHOULD be encoded as UTF-8; other character encoding schemes MAY be supported, but are discouraged.
```

File: javascript-mjs/draft.md [hunk 2]
```diff
@@ -149,51 +149,43 @@ Implementations need to determine a character encoding scheme in order to decode
...
 
 The charset parameter provides a means to specify the character encoding scheme of binary source text.  Its value MUST match the mime-charset production defined in {{RFC2978}}, section 2.3, and SHOULD be a registered charset {{CHARSETS}}.  An illegal value is a value that does not match that production.
 
-## Character Encoding Scheme Detection
-
-It is possible that implementations cannot interoperably determine a single character encoding scheme simply by complying with all requirements of the applicable specifications.  To foster interoperability in such cases, the following algorithm is defined.
+The charset parameter is only used when processing a Script goal source; Module goal sources MUST always be processed as UTF-8.
 
-Implementations apply this algorithm until a single character encoding scheme is determined.
+## Character Encoding Scheme Detection
```
Comment (annevk) [inline javascript-mjs/draft.md:160]:
```suggestion
## Character Encoding Scheme Detection for Script Goal Sources
```

**Comment by annevk** [comment]
Looks okay otherwise btw.

File: javascript-mjs/draft.md [hunk 3]
```diff
@@ -115,7 +115,7 @@ This document updates the ECMAScript media types, replacing the existing registr
 
 # Introduction
 
-This memo describes media types for the JavaScript and ECMAScript programming languages.  Refer to "Brief History" and "Overview" in {{ECMA-262}} for background information on these languages.  This document updates the descriptions and registrations for these media types to reflect existing usage on the Internet.
+This memo describes media types for the JavaScript and ECMAScript programming languages.  Refer to the section "Introduction" in {{ECMA-262}} for background information on these languages.  This document updates the descriptions and registrations for these media types to reflect existing usage on the Internet.
```
Comment (codehag) [inline javascript-mjs/draft.md:118]: I think you can leave "Overview", as it is still present in the specification: https://tc39.es/ecma262/#sec-overview

File: javascript-mjs/draft.md (hunk 2 above)
Comment (linuxwolf) [reply javascript-mjs/draft.md:160]: The current text doesn't say where the charset parameter is coming from -- whether that is from transport metadata (e.g., `Content-Type`) or elsewhere (e.g., `<script type='...' >`).  Is that sufficiently open-ended?

Comment (annevk) [reply javascript-mjs/draft.md:160]: I guess so, though either way HTML will need to keep its own algorithm I suspect.

Comment (linuxwolf) [reply javascript-mjs/draft.md:160]: I also suspect that's true, and that should be ok.

**Pull request merged**